### PR TITLE
CFD-286 Homepage : Featured block

### PR DESCRIPTION
**Pull request by HelenaEksler**
#472

### CODE REVIEW

**Comment by HelenaEksler** [comment]
selection_205

@nader77 Review this, please.

File: capacity4more/modules/c4m/content/c4m_content_featured/c4m_content_featured.module [hunk 1]
```diff
@@ -5,3 +5,74 @@
...
+/**
+ * Implements hook_ctools_plugin_directory().
+ */
+function c4m_content_featured_ctools_plugin_directory($module, $plugin) {
+  if ($module == 'ctools') {
+    return 'plugins/' . $plugin;
+  }
+}
+
+function c4m_content_featured_render_featured_blocks() {
```
Comment (nader77) [inline capacity4more/modules/c4m/content/c4m_content_featured/c4m_content_featured.module]: Add docs.

File: capacity4more/modules/c4m/content/c4m_content_featured/c4m_content_featured.module [hunk 2]
```diff
@@ -5,3 +5,74 @@
...
+
+/**
+ * Implements hook_ctools_plugin_directory().
+ */
+function c4m_content_featured_ctools_plugin_directory($module, $plugin) {
+  if ($module == 'ctools') {
+    return 'plugins/' . $plugin;
+  }
+}
+
```
Comment (nader77) [inline capacity4more/modules/c4m/content/c4m_content_featured/c4m_content_featured.module]: Add docs.

File: capacity4more/modules/c4m/content/c4m_content_featured/c4m_content_featured.module [hunk 3]
```diff
@@ -5,3 +5,74 @@
...
+ * Implementation of hook_query_TAG_alter
+ */
+function c4m_content_featured_query_random_alter($query) {
+  $query->orderRandom();
+}
+
+/**
+ * Implements hook_entity_info_alter().
+ *
+ * Add "Upcoming event" view mode.
```
Comment (nader77) [inline capacity4more/modules/c4m/content/c4m_content_featured/c4m_content_featured.module]: "Upcoming event" => "Featured"

File: capacity4more/modules/c4m/content/c4m_content_featured/c4m_content_featured.module [hunk 4]
```diff
@@ -5,3 +5,74 @@
...
+function c4m_content_featured_preprocess_node(&$variables) {
+  if ($variables['view_mode'] == 'featured_block') {
+    $variables['theme_hook_suggestions'][] = 'node__' . $variables['node']->type . '__featured_block';
+  }
+  $preprocess_function = "c4m_content_featured_preprocess_node__{$variables['node']->type}";
+  if (function_exists($preprocess_function)) {
+    $preprocess_function($variables);
+  }
+}
+
```
Comment (nader77) [inline capacity4more/modules/c4m/content/c4m_content_featured/c4m_content_featured.module]: Add docs.

**Comment by nader77** [comment]
@HelenaEksler [In the wiki page](https://wiki.amplexor.com/confluence/display/C4DNG/Home+page+-+Anonymous+user) it says "Anonymous: fixed untill new content has been added/updated, "
Looks like the featured block should behave differently for guests.

File: capacity4more/modules/c4m/content/c4m_content_featured/c4m_content_featured.module (hunk 2 above)
Comment (HelenaEksler) [inline capacity4more/modules/c4m/content/c4m_content_featured/c4m_content_featured.module]: Docs one line below

File: capacity4more/modules/c4m/content/c4m_content_featured/c4m_content_featured.module (hunk 4 above)
Comment (HelenaEksler) [inline capacity4more/modules/c4m/content/c4m_content_featured/c4m_content_featured.module]: Docs one line below

**Comment by HelenaEksler** [comment]
@nader77  Added docs. Made differences for anonymous and authenticated user due to wiki.

**Comment by nader77** [comment]
Installed and reviewed, Looks good. 

@zero2one 
We didn't work at the design at all, Only functionality, Should we at least make the featured blocks equal in height as [requested in the Wireframe](https://wiki.amplexor.com/confluence/display/C4DNG/Home+page+-+Anonymous+user), Or is it good like this ?

**Comment by HelenaEksler** [comment]
Closed in favor of #489

**Pull request closed without merging**